crates/core/src/host/v8/de.rs

```diff
@@ -0,0 +1,420 @@
+#![allow(dead_code)]
+
+use super::error::{exception_already_thrown, ExceptionThrown};
+use super::from_value::{cast, FromValue};
+use core::fmt;
+use core::iter::{repeat_n, RepeatN};
+use core::mem::MaybeUninit;
+use derive_more::From;
+use spacetimedb_sats::de::{
+    self, ArrayVisitor, DeserializeSeed, NoneAccess, ProductVisitor, SliceVisitor, SomeAccess, SumVisitor,
+};
+use spacetimedb_sats::{i256, u256};
+use std::borrow::{Borrow, Cow};
+use v8::{Array, Global, HandleScope, Local, Object, Uint8Array, Value};
+
+/// Deserializes from V8 values.
+pub(super) struct Deserializer<'a, 's> {
+    common: DeserializerCommon<'a, 's>,
+    input: Local<'s, Value>,
+}
+
+impl<'a, 's> Deserializer<'a, 's> {
+    /// Creates a new deserializer from `input` in `scope`.
+    pub fn new(scope: &'a mut HandleScope<'s>, input: Local<'_, Value>, key_cache: &'a mut KeyCache) -> Self {
+        let input = Local::new(scope, input);
+        let common = DeserializerCommon { scope, key_cache };
+        Deserializer { input, common }
+    }
+}
+
+/// Things shared between various [`Deserializer`]s.
+///
+/// The lifetime `'s` is that of the scope of values deserialized.
+struct DeserializerCommon<'a, 's> {
+    /// The scope of values to deserialize.
+    scope: &'a mut HandleScope<'s>,
+    /// A cache for frequently used strings.
+    key_cache: &'a mut KeyCache,
+}
```

Comment:
What's the lifetime `'a`? And can we just use words to name these lifetimes, like `'scope` instead of `'s`? Fine if `'a` doesn't have a meaningful name, but a comment describing the meaning of a single-letter variable is a little :thinking: .